### PR TITLE
change on catalog.sql in case big information_schema

### PR DESCRIPTION
**Pull request by mauricioobgo**
resolves #

### Description

This optimize catalog.sql when query larger information_schema in order to create faster documentation in dbt

### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-snowflake/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

**Comment by Fleid** [comment]
I feel like this is useful but without context it's hard to gauge.
Do we have an issue for it? Have you benchmarked the gain in performance? Do we know of any side effects?

**Comment by mauricioobgo** [comment]
> I feel like this is useful but without context it's hard to gauge. Do we have an issue for it? Have you benchmarked the gain in performance? Do we know of any side effects?

- No, there is no currently an issue.
- The first cte was performing just a select on all the information schema as it is the columns cte and you only filtering at the end, hence your doing a select all of twice in the same information schema before filtering with the join but if you filter before the two cte at the end you are joining less amount of data therefore an increase in performance. This reduce or catalog to run in 30 min from 3 hrs, since our schemas have in avg 10Th tables.
- Side effects, would be not looping the entire information schema within the database just the ones you have.

File: .changes/unreleased/Features-20220815-203822.yaml
Comment (mikealfare) [inline .changes/unreleased/Features-20220815-203822.yaml:1]: Please delete this changelog as it's a duplicate and it points to a non-existent issue.

Comment (mauricioobgo) [reply .changes/unreleased/Features-20220815-203822.yaml:1]: Deleted @mikealfare
Thank you

**Review by mikealfare — APPROVED** [review]

**Pull request merged**